### ISSUE TITLE
Node attributes in _cat/nodes

### ISSUE THREAD

**Issue by suyograo** [elastic/elasticsearch#8000]
It would be useful if `_cat/nodes` can output node attributes like rack_id, aws_zone and other user defined attributes. If users have a large cluster with nodes in multiple AZ's or racks, it would be great to have this info in `_cat/nodes` to understand the topology. 

Even if not displayed by default, we could allow users to pass in `nodes?h=rack_id`

Example:

```
host                     ip          heap.percent ram.percent load node.role master name rack_id
Suyogs-MacBook-Pro.local 192.168.1.6            0                  d         *      Jann rack_1
Suyogs-MacBook-Pro.local 192.168.1.6            0                  d         *      Hoder rack_2
```

**Comment by drewr** [comment]
Sorry to just be noticing this, but I caught the email from @metadave's PR.

This has a similar problem to `_cat/fielddata` that #10249 is meant to address, namely that column names should never be dynamic. We always denormalize instead. See `_cat/shards` for example.

In this case, since node attributes are such a small part of a node identity, I wonder if it doesn't make sense to have a new API, `_cat/nodeattrs` or somesuch?

```
% GET /_cat/nodeattrs
node    attr.name     attr.value
n1      zone          zone2323
n2      zone          zone8282
```

The default could have more info, maybe `host` and `port`.

**Comment by metadave** [comment]
+1, I like this looks

**Comment by metadave** [comment]
Node attributes are available in `_cat/nodeattrs` via #12534
